### DISCUSSION TITLE
v3 upgrade issue (`Class "Filament\PluginServiceProvider" not found`)

### DISCUSSION THREAD

**Discussion by jdarkins** [filamentphp/filament#9924] [Help] [answered]
Hi,

I'm trying to upgrade a project from v2 -> v3 but running into some issues during the automatic upgrade process. 

I run the `composer require filament/upgrade:"^3.0-stable" -W --dev` command succesfully with everything showing green.

I then run the `vendor/bin/filament-v3` command and follow the composer require commands listed. I had some dependency issues so I did the following:

- `composer require jeffgreco13/filament-breezy:^2.2 -W --no-update` as the version I had was locked to filament 2. 
- `"ryangjchandler/filament-progress-column": "^0.3.2",` removed as this doesn't look like it supports v3 yet
- `"livewire/livewire": "^2.12.2",` was removed after looking at previous discussions when running into issues there.

I then run `composer update` and all the packages update with no issues, but run into the following:

```
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

In FilamentImpersonateServiceProvider.php line 14:

  Class "Filament\PluginServiceProvider" not found


Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
```

Any suggestions will be appreciated, cheers!

**Comment by leandrocfe** [comment, marked as answer]
Did you update the filament impersonate version to ˆ3.0?
https://github.com/stechstudio/filament-impersonate/tags

**Reply by jdarkins** [reply]
Ahhh that's the one, thank you! I didn't set this project up originally so I didn't realise it was referring to another dependency, I thought it was an issue with a core file
